### ISSUE TITLE
Suggestion to improve TRY003 rule

### ISSUE THREAD

**Issue by Booplicate** [astral-sh/ruff#14398]
### Issue
I want to propose to change how TRY003 works with some of the builtin exceptions. While I like how the rule applies to custom exceptions and "base" `Exception`, I think it's incorrect in the case of errors like `TypeError` and `ValueError`. It's a common pattern and is widely recommended to use those exceptions when type or value don't match the expectations.
[Here](https://github.com/astral-sh/ruff/issues/2246#issuecomment-1406479505) you suggest to define a subclass of `Exception`, but not only defining a separate exception for _every_ case is redundant and would bloat the code, I'm pretty sure using `ValueError` would _more correct_.

You already ignore `NotImplementedError`, which was a decision in the right direction! It doesn't make sense to subclass it every time. I think the same applies for at least `ValueError` and `TypeError`.

### Solution

Currently you have such code to ignore `NotImplementedError`:
https://github.com/astral-sh/ruff/blob/abb34828bdbce0eff28d4350b8f188436e57be78/crates/ruff_linter/src/rules/tryceratops/rules/raise_vanilla_args.rs#L71-L78
sadly it seems that `match_builtin_expr` can only take a single `symbol`. While we could call it multiple times, I think it'd be slightly ineffective? A better approach might be changing this method (or defining a new one), so it can take an array of symbols to match against.

Related issues: #2246 #6528

Sample code:
```py
def compute(a: int, b: int) -> int:
     if a <= 0:
         raise ValueError("a must be > 0")  # Ruff: Avoid specifying long messages outside the exception class [TRY003]
     if not b:
         raise ValueError("b must be != 0")  # Ruff: Avoid specifying long messages outside the exception class [TRY003]
     return a + b
```

`ruff` command: `ruff check --no-fix`

`ruff` version: 0.7.4

`ruff` settings:
```toml
[tool.ruff.lint]
select = ["TRY003"]
```

**Comment by edwinvehmaanpera** [comment]
Full configurability would be ideal, but if not then I would suggest also adding `AssertionError` to the ignored exceptions.

**Comment by bedlamzd** [comment]
Another candidate is `ExceptionGroup`

**Comment by k4lizen** [comment]
I came to raise an issue about `AssertionError` and found this. I think its completely uncontraversial that `AssertionError` should be added alongside `NotImplementedError`.

**Comment by wlach** [comment]
I'd also add `RuntimeError` into the mix, which IME is almost always bespoke.

**Comment by jack-volantautonomy** [comment]
I agree with this thread that the built ins should probably not get flagged on this check as it's very struct and should require bespoke handling, however `ruff` is just meant to replicate `tryceratops` behaviour as @charliermarsh has mentioned previously so I think this is more a change/suggestion that should be proposed to `tryceratops` instead. Will start a suggestion thread there.

**Comment by jack-volantautonomy** [comment]
It appears that `tryceratops` isn't actively maintained, but the idea of ignoring this check for built-ins has been proposed before and is generally supported. Could we deviate from the replicating this linter identically and improve the ruff `TRY003` check by ignoring anything that provides a message to a built-in exception?

**Comment by wlach** [comment]
> It appears that `tryceratops` isn't actively maintained, but the idea of ignoring this check for built-ins has been proposed before and is generally supported. Could we deviate from the replicating this linter identically and improve the ruff `TRY003` check by ignoring anything that provides a message to a built-in exception?

FWIW I've kind of moved towards the opinion that this rule does more harm than good overall. There's just so many legitimate cases for providing a custom message for an exception (one common pattern I've seen is ValidationError type errors when doing schema checking) that I don't feel like a rule like this really makes sense, even if it occasionally catches the occasional anti-pattern/true positive.
